jenkins/docker/Jenkinsfile

```diff
@@ -41,25 +53,77 @@ pipeline {
+                            '''
+                        }
+                    } else {
+                        def CREDENTIAL_ID = "jenkins-staging-docker-staging-credential"
+                        withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                            sh '''
+                                set -ex
+                                echo Login to $CREDENTIAL_ID
+                                docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD && eval $DOCKER_BUILD_SCRIPT_WITH_COMMANDS
+                            '''
+                        }
+                    }
+                }
+            }
+            post() {
+                always {
+                    script {
+                        postCleanup()
+                        sh "docker logout && docker image prune -f --all"
                     }
-                    sh "echo Account: $CREDENTIAL_ID"
-                    withCredentials([usernamePassword(credentialsId: CREDENTIAL_ID, usernameVariable: 'DOCKER_USERNAME', passwordVariable: 'DOCKER_PASSWORD')]) {
+                }
+            }
+        }
+        stage('docker-build-ecr') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-Ubuntu2004-X64-m52xlarge-Docker-Builder'
+                    image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+                    args '-u root -v /var/run/docker.sock:/var/run/docker.sock'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    withAWS(role: 'abhng-ecr-role', roleAccount: "${ABHNG_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-session') {
                         sh '''
-                            set -ex
-                            echo Login to $CREDENTIAL_ID
-                            docker logout && docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD && eval $DOCKER_BUILD_SCRIPT_WITH_COMMANDS
+                            curl -L https://github.com/google/go-containerregistry/releases/latest/download/go-containerregistry_Linux_x86_64.tar.gz \\
+                                -o go-containerregistry.tar.gz
+                            tar -zxvf go-containerregistry.tar.gz
+                            chmod +x gcrane
+                            mv gcrane /usr/local/bin/
+                            docker logout
+                            aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/a5n0l3m1
+                            gcrane cp opensearchstaging/opensearch:latest public.ecr.aws/m0o1u6w1/opensearch:latest
+                            gcrane cp public.ecr.aws/a5n0l3m1/opensearch-test:1.3.0 public.ecr.aws/a5n0l3m1/opensearch-copy:1.3.0
```

Comment:
Seems like these should be parameterized, why the double copy?

Reply:
was just testing out a few things there